packages/grid/_modules_/grid/models/api/gridRowsMetaApi.ts

```diff
@@ -11,4 +11,11 @@ export interface GridRowsMetaApi {
    * @ignore - do not document.
    */
   unstable_getRowHeight: (id: GridRowId) => number;
+  /**
+   * Updates the height of a row.
+   * @param {GridRowId} id The id of the row.
+   * @param {number} height The new height.
+   * @ignore - do not document.
+   */
+  unstable_setRowHeight: (id: GridRowId, height: number) => void;
```

Comment:
Wouldn't this API be public? I was left with the impression that the reason we are adding it is so that devs can target specific rows manually. If that is the case then it shouldn't have the `unstable_` prefix. 
If it is for internal use only I would argue that we don't need it (or at least we don't need it at this point in time).

Reply:
Same feeling
This API only goal is to be used by the user, so making it private kind of kill this usecase.

Reply:
It will be used when we add the the resize handle. Since we don't know if `setRowHeight` will work for that purpose and it won't require heavy changes I would keep private. But it should be public once we know it's stable.

Reply:
Exactly what @m4theushw said.

One more thing I was thinking of is that we might change its signature to `setRowsHeight` to support resizing multiple rows at once. This way `hydrateRowsMeta()` will get called once per resize, not once per row per resize.
But maybe I should implement it like that in the first place?
```tsx
unstable_setRowsHeight: (Array<{ id: GridRowId, height: number }>) => void;
```

What do you think?

Reply:
Is there a use case for resizing multiple rows at once? Maybe only through a button. If there's one, `unstable_setRowsHeight` could call the `unstable_setRowHeight` but have an argument to only `hydrateRowsMeta` after the last row. I remember we had something similar for postponing the call to `applyFiters`.

Reply:
For `updateColumns` we are doing the opposite : `updateColumn` calls `updateColumns`
And I proposed to remove `updateColumn` because I don't think having both methods is worth it.

____

```ts
unstable_setRowsHeight: (rows: Array<{ id: GridRowId, height: number }>) => void;
```

Is there an advantage of using `Array<X>` over `X[]`

```ts
unstable_setRowsHeight: (rows: { id: GridRowId, height: number }[]) => void;
```

Reply:
> Is there a use case for resizing multiple rows at once?

I have no idea, but you can do that in Google spreadsheet for example.
I'm just trying to avoid future breaking changes (if we want to ditch `unstable_` prefix now)

> For updateColumns we are doing the opposite : updateColumn calls updateColumns

Yes, that's the reason I'm proposing the array signature.
Also, I think it's more elegant than `setRowHeight(hydrateRowsMeta)`, because we don't need to explain what is `hydrateRowsMeta` and when to use it.

> Is there an advantage of using Array<X> over X[] ?

AFAIK those two signatures are equal, so it doesn't matter

Reply:
> It will be used when we add the the resize handle. Since we don't know if `setRowHeight` will work for that purpose and it won't require heavy changes I would keep private. But it should be public once we know it's stable.

I'm a bit confused. It seems like we are adding an API method to hypothetically cover a case that may or may not exist. I was left with the impression that this new API method will be used eventually to support row resizing (similar to column resizing) but I don't know if this is even a feature we want to add eventually.
I would be in favor of starting with a very narrow scope -> resize a target row - and make that public.

Reply:
We have two features that could leverage `setRowHeight`: #3801 and #417. This PR could be the first step to #417. Any spreadsheet app allows to automatically adjust the row height to the content height by double clicking the row. Instead of a full dynamic height feature, we could allow the same thing in the DataGrid. Users have been encouraged to use [this demo](https://github.com/mui-org/material-ui-x/issues/417#issuecomment-782772202) but you can only see the full content on hover.

Reply:
I'm not about always adding an imperative API to every feature. I think it should be on a case-by-case basis, otherwise, we will end up with too much public API, like it was before.
Either way - I'm not saying we shouldn't add that API method but rather just remove the `unstable_` prefix and keep it as it is - update only 1 row.

Reply:
On today's meeting we came to a conclusion that we are sure about stability of this API and we'll keep `unstable_` for now